enterprise/ha/src/test/java/org/neo4j/kernel/ha/id/HaIdGeneratorFactoryTest.java

```diff
@@ -200,6 +205,39 @@ public void shouldTranslateComExceptionsIntoTransientTransactionFailures() throw
         generator.nextId();
     }
 
+    @Test
+    public void shouldNotUseForbiddenMinusOneIdFromIdBatches() throws Exception
+    {
+        // GIVEN
+        long[] defragIds = {3, 5};
+        int size = 10;
+        long low = IdGeneratorImpl.INTEGER_MINUS_ONE - size/2;
+        IdRange idRange = new IdRange( defragIds, low, size );
+
+        // WHEN
+        IdRangeIterator iterartor = new IdRangeIterator( idRange );
+
+        // THEN
+        for ( long id : defragIds )
+        {
+            assertEquals( id, iterartor.next() );
+        }
+
+        int expectedRangeSize = size - 1; // due to the forbidden id
```

Comment:
This concerns me. This means that the length of an `idRange` can't be trusted. And I wonder if that will cause the assumptions in for example `HaIdGeneratorFactor.getNumberOfIdsInUse` to break.